### ISSUE TITLE
getDistributedObjects does not return CPSubsystem  objects

### ISSUE THREAD

**Issue by pavelorehov** [hazelcast/hazelcast#17135]
In Hazelcast 4.0.1 getDistributedObjects does not return locks and semaphores and there is also no getDistributedObjects  method to get them from CP Subsystem.

So no way to get existing distributed objects in CP Subsystem.

**Comment by mmedenjak** [comment]
Hi @pavelorehov ! Thank you for your report. It's indeed an oversight on our side. Adding it to the current milestone so we take a look. cc @mdogan @petrpleshachkov

**Comment by mmedenjak** [comment]
@mdogan @petrpleshachkov any ideas if we could easily add this in 4.1?

**Comment by mdogan** [comment]
I don't think we can integrate this into `HazelcastInstance. getDistributedObjects()`. Instead we can introduce a new method under `CPSubsystem`.  But it may not have the same guarantees of the former method; like returning all CP objects created in cluster. We may only return locally known objects. Otherwise it needs a coordination between members of multiple CP groups.

**Comment by mmedenjak** [comment]
Thank you @mdogan , moving from 4.1 and adding the enhancement label.

**Comment by pavelorehov** [comment]
Any updates on the delivery of the missing functionality ? For us it is more regression defect rather than enhancement as there is NO way to get list and status of new CP locks and semaphores.

**Comment by prabhushan** [comment]
This is absolutely necessary

**Comment by FloLie** [comment]
Any updates on this? As @pavelorehov said, this is not an Enhancement, but makes it impossible to detect memory leaks, clean up locks and semaphores and so on. I can't believe this has been open for almost 3 years now.

**Comment by pavelorehov** [comment]
Trying almost year later, any updates ?
